src/platform/Linux/ConnectivityUtils.cpp

```diff
@@ -310,6 +310,90 @@ CHIP_ERROR ConnectivityUtils::GetInterfaceHardwareAddrs(const char * ifname, uin
     return err;
 }
 
+CHIP_ERROR ConnectivityUtils::GetInterfaceIPv4Addrs(const char * ifname, uint8_t & size, NetworkInterface * ifp)
+{
+    CHIP_ERROR err;
+    struct ifaddrs * ifaddr = nullptr;
+
+    if (getifaddrs(&ifaddr) == -1)
+    {
+        ChipLogError(DeviceLayer, "Failed to get network interfaces");
+        err = CHIP_ERROR_READ_FAILED;
+    }
+    else
+    {
+        uint8_t index = 0;
+
+        for (struct ifaddrs * ifa = ifaddr; ifa != nullptr; ifa = ifa->ifa_next)
+        {
+            if (ifa->ifa_addr->sa_data == NULL)
+            {
+                continue;
+            }
+
+            if (ifa->ifa_addr && ifa->ifa_addr->sa_family == AF_INET)
+            {
+                if (strcmp(ifname, ifa->ifa_name) == 0)
+                {
+                    void * addPtr = &((struct sockaddr_in *) ifa->ifa_addr)->sin_addr;
+
+                    memcpy(ifp->Ipv4AddressesBuffer[index], addPtr, kMaxIPv4AddrSize);
+                    ifp->Ipv4AddressSpans[index] = ByteSpan(ifp->Ipv4AddressesBuffer[index], kMaxIPv4AddrSize);
+                    index++;
```

Comment:
What happens if there are enough interfaces that this runs over the length of `ifp->Ipv4AddressesBuffer`?

Reply:
We should skip the rest of interfaces.